src/mon/FSCommands.cc

```diff
@@ -270,12 +271,24 @@ class SetHandler : public FileSystemCommandHandler
         ss << "may not have more than " << MAX_MDS << " MDS ranks";
         return -EINVAL;
       }
+
       fsmap.modify_filesystem(
           fs->fscid,
           [n](std::shared_ptr<Filesystem> fs)
       {
         fs->mds_map.set_max_mds(n);
       });
+
+      unsigned in = fs->mds_map.get_num_in_mds();
+      mds_rank_t trim = std::min((mds_rank_t)in, oldmax);
+      for (mds_rank_t i = trim; i > n; i--) {
+	MDSMap::mds_info_t mi = fs->mds_map.get_info(i-1);
+	ss << "Setting mds." << mi.name << " to standby. ";
+	fsmap.modify_daemon(mi.global_id, [] (MDSMap::mds_info_t *info) {
+	    info->state = MDSMap::STATE_STOPPING;
+	});
+      }
+
```

Comment:
If we deactivate mds automatically when decreasing max_mds. we show add code that restart the deactivating process when the stopping mds failover. this is importance specially when deactivating multiple MDS at the same time. Otherwise, mds cluster may end up with status like: (rank0 active, rank1 stopped, rank2 active)